components/resolved_ts/src/advance.rs

```diff
@@ -275,7 +277,12 @@ impl LeadershipResolver {
     // This function broadcasts a special message to all stores, gets the leader id
     // of them to confirm whether current peer has a quorum which accepts its
     // leadership.
-    pub async fn resolve(&mut self, regions: Vec<u64>, min_ts: TimeStamp) -> Vec<u64> {
+    pub async fn resolve(
+        &mut self,
+        regions: Vec<u64>,
+        min_ts: TimeStamp,
+        timeout: Option<Duration>,
```

Comment:
`timeout` is actually the timeout of inner grpc calls, in the worst case, `resolve` may take about `2*timeout`.

Reply:
Do you mean `get_tikv_client` takes a `timeout` duration but return `Ok<...>`, then the RPC takes another `timeout` duration? It's rare to be such worse IMO, but we may replace the `tokio::time::timeout` with `tokio::time::timeout_at` in the future.

Reply:
Yes, I just put a comment here, the PR LGTM.